src/ll/ll-parsing-table.js

```diff
@@ -175,20 +175,17 @@ export default class LLParsingTable {
         table[lhsSymbol] = {};
       }
 
-      // All productions goes under the terminal column, if
-      // this terminal is not epsilon. Otherwise, an ε-production
-      // goes under the columns from the Follow set of LHS.
-
-      let set = !production.isEpsilon()
-        ? this._setsGenerator.firstOfRHS(rhs)
-        : this._setsGenerator.followOf(lhs);
-
-      const keys = Object.keys(set);
-
-      // Got derived epsilon through indirect production, fallback to
-      // Follow(LHS). Example: A: B; B: ε
-      if (keys.length === 1 && GrammarSymbol.isEpsilon(keys[0])) {
-        set = this._setsGenerator.followOf(lhs);
+      // All productions goes under the terminal column
+      // of the First(RHS) terminals.
+      let set = this._setsGenerator.firstOfRHS(rhs);
+
+      // If First(RHS) has an epsilon terminal it means
+      // that we must check also under the Follow(LHS) column.
+      // (because this production can be eliminated by epsilon).
+      if (set[EPSILON]) {
```

Comment:
Can you replace please with:
```
if (set.hasOwnProperty(EPSILON)) {
  ...
}
```

(later we may need to refactor to using  actual Set JS object instead of plain objects)